projects/vdk-control-cli/src/vdk/internal/control/command_groups/job/create.py

```diff
@@ -63,15 +69,45 @@ def __create_cloud_job(self, team: str, name: str) -> None:
             f"Data Job with name {name} created and registered in cloud runtime by Control Service."
         )
 
-    def __create_local_job(self, team: str, name: str, job_path: str) -> None:
-        sample_job = self.__vdk_config.sample_job_directory
+    def __create_local_job(
+        self, team: str, name: str, job_path: str, sample_job_directory
+    ) -> None:
+        sample_job = self.__create_sample_job_dir(sample_job_directory)
         log.debug(f"Create sample job from directory: {sample_job} into {job_path}")
         cli_utils.copy_directory(sample_job, job_path)
         local_config = JobConfig(job_path)
         if not local_config.set_team_if_exists(team):
             log.warning(f"Failed to write Data Job team {team} in config.ini.")
         log.info(f"Data Job with name {name} created locally in {job_path}.")
 
+    def __create_sample_job_dir(self, sample_job_directory):
+        """
+        This method generates the directory for the sample job which will be created when invoking `vdk create`.
+        Its control flow is as follows:
+         - if the configuration variable VDK_CONTROL_SAMPLE_JOB_DIRECTORY is set, it takes precedent over
+         everything else
+         - next, if a sample_job_module is passed to the method by its invoking class, it takes precedent next; the
+         purpose of this is to allow plugins and extensions to set a new default sample job, but still not overwrite
+         one which was set by the user
+         - finally, the default sample job is used
+
+        :param sample_job_module: a Python module containing the files for a sample job
+        :return: an absolute path in the current environment where the sample job is so the files located there can
+        be copied over to the target directory when creating a new job
+        """
+        if self.__vdk_config.sample_job_directory:
+            return self.__vdk_config.sample_job_directory
+
+        if not sample_job_directory:
+            import vdk.internal.control.job.sample_job
+
+            template_module_path = vdk.internal.control.job.sample_job.__path__._path[0]
+            sample_job_dir = os.path.abspath(template_module_path)
+        else:
+            template_module_path = sample_job_directory.__path__._path[0]
+            sample_job_dir = os.path.abspath(template_module_path)
+        return sample_job_dir
```

Comment:
that doesn't seem correct?

You've removed draft but it seems still a draft . Have you forgotten to push a commit

Reply:
which part does not seem correct?

Reply:
hmmm, I thought sample_job_directory would be of type "str" or pathlib.Path ? 

That's why that seems incorrect ...